cherrysrv/client_test.go

```diff
@@ -71,3 +83,68 @@ func TestClient(t *testing.T) {
 
 	}
 }
+
+// TestMultipleClients
```

Comment:
These test cases tend to deadlock and need to be redesigned.

Reply:
Even if the code is valuable, I think the your code to solve this issue is too complex. I need some time to think about it. 

Could you do a PR for the testing only? This is something I'm happy to apply.

Reply:
Where did we leave off on this? @kdedon did you intend to break this into 2 pieces? I hate to have this dangling out here for so long.